java-design-patterns-master/data-transfer-object/src/main/java/com/iluwatar/datatransfer/CustomerClientApp.java

```diff
@@ -0,0 +1,80 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.iluwatar.datatransfer;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The Data Transfer Object pattern is a design pattern in which an data transfer object is used to
+ * serve related information together to avoid multiple call for each piece of information.
+ *
+ * <p>In this example, ({@link CustomerClientApp}) as as customer details consumer i.e. client to
+ * request for customer details to server.
+ *
+ * <p>CustomerResource ({@link CustomerResource}) act as server to serve customer information. And
+ * The CustomerDto ({@link CustomerDto} is data transfer object to share customer information.
+ */
+public class CustomerClientApp {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CustomerClientApp.class);
+
+  /**
+   * Method as act client and request to server for details.
+   *
+   * @param args program argument.
+   */
+  public static void main(String[] args) {
+    var customerOne = new CustomerDto("1", "Kelly", "Brown");
+    var customerTwo = new CustomerDto("2", "Alfonso", "Bass");
+    var customers = new ArrayList<>(List.of(customerOne, customerTwo));
+
+    var customerResource = new CustomerResource(customers);
+
+    LOGGER.info("All customers:-");
+    var allCustomers = customerResource.getAllCustomers();
+    printCustomerDetails(allCustomers);
+
+    LOGGER.info("----------------------------------------------------------");
+
+    LOGGER.info("Deleting customer with id {1}");
+    customerResource.delete(customerOne.getId());
+    allCustomers = customerResource.getAllCustomers();
+    printCustomerDetails(allCustomers);
+
+    LOGGER.info("----------------------------------------------------------");
+
+    LOGGER.info("Adding customer three}");
+    var customerThree = new CustomerDto("3", "Lynda", "Blair");
+    customerResource.save(customerThree);
+    allCustomers = customerResource.getAllCustomers();
+    printCustomerDetails(allCustomers);
+  }
+
+  private static void printCustomerDetails(List<CustomerDto> allCustomers) {
+    allCustomers.forEach(customer -> LOGGER.info(customer.getFirstName()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code contains a potential information leakage in the log handling for the following call(s): &#39;CustomerDto.getFirstName()&#39; (line 78). You are handling this log data through: &#39;Logger.info()&#39; (line 78).  Please use your organization-specific sensitive information handling loggers or redaction. [Learn more about information leakage](https://cwe.mitre.org/data/definitions/532.html) (external link).